src/qinfer/resamplers.py

```diff
@@ -299,30 +299,25 @@ def __call__(self, model, particle_dist,
 
         idxs_to_resample = np.arange(n_particles, dtype=int)
 
-        # Preallocate js and mus so that we don't have rapid allocation and
-        # deallocation.
-        js = np.empty(idxs_to_resample.shape, dtype=int)
-        mus = np.empty(new_locs.shape, dtype=l.dtype)
-
         # Loop as long as there are any particles left to resample.
         n_iters = 0
 
         # Draw j with probability self.particle_weights[j].
         # We do this by drawing random variates uniformly on the interval
         # [0, 1], then see where they belong in the CDF.
-        js[:] = cumsum_weights.searchsorted(
+        js = cumsum_weights.searchsorted(
             np.random.random((idxs_to_resample.size,)),
             side='right'
         )
 
+        # Set mu_i to a x_j + (1 - a) mu.
+        # FIXME This should use particle_dist.particle_mean
+        mus = a * l[js,:] + (1 - a) * mean
+
```

Comment:
I have the computation of `mus` out of the loop since it's inputs don't change in the loop. In a benchmarking example, which was tailored to require a lot of iterations, this gave a slight performance increase of roughly 15%. 

The main reason I moved this out is that this should actually use `ParticleDistribution.particle_mean` since it should be considered as a particle mean over 2 particles with weights `a` and `1 - a`. It doesn't matter for the standard mean function, but for custom mean functions this might give wrong results.

However, I would keep it the way it is since
- using `particle_mean` as it is requires broadcasting the `mean`, making `n_particles` copies
- for circular means, there is a workaround by disabling postselection and enabling canonicalization (see [here](https://github.com/dseuss/qinfer-examples/blob/master/circular_statistics.ipynb))